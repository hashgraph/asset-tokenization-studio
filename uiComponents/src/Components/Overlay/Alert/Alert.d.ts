import type { AlertProps as ChakraAlertProps } from "@chakra-ui/alert";
import type { alertAnatomy as ChakraAlertParts } from "@chakra-ui/anatomy";
import type { ComponentWithAs, SystemStyleObject } from "@chakra-ui/system";
import React from "react";
import type { BaseMultiStyleConfiguration, PartsStyleInterpolation } from "@Theme/Components/BaseMultiStyleConfiguration";
import type { PhosphorIconProps } from "@/Components/Foundations/PhosphorIcon";
type UnionKeys<T> = T extends T ? keyof T : never;
type StrictUnionHelper<T, TAll> = T extends any ? T & Partial<Record<Exclude<UnionKeys<TAll>, keyof T>, never>> : never;
type StrictUnion<T> = StrictUnionHelper<T, T>;
export interface BaseAlertProps extends Omit<ChakraAlertProps, "children" | "title" | "status"> {
    isInline?: boolean;
    onClose?(): void;
    showIcon?: boolean;
    status?: AlertStatus;
}
export interface CustomChildrenAlertProps {
    children: React.ReactNode;
}
export interface TitleDescriptionAlertProps {
    title?: string;
    description?: string;
}
export type AlertProps = BaseAlertProps & StrictUnion<TitleDescriptionAlertProps | CustomChildrenAlertProps>;
export declare const alertPartsList: Array<(typeof ChakraAlertParts.keys)[number] | "closeBtn" | "contentContainer">;
type Parts = typeof alertPartsList;
export type AlertStatus = "success" | "warning" | "error" | "info" | "loading";
export interface AlertThemeConfiguration extends BaseMultiStyleConfiguration<Parts> {
    baseStyle: (({ status }: {
        status: AlertStatus;
    }) => Partial<AlertThemeStyle>) | PartsStyleInterpolation<Parts>;
}
type AlertThemeStyle = Record<Parts[number], SystemStyleObject> & {
    icon: PhosphorIconProps;
};
export declare const Alert: ComponentWithAs<"div", AlertProps>;
export {};
