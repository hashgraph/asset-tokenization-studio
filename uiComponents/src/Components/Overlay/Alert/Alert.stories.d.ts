import type { StoryFn } from "@storybook/react";
import React from "react";
import { Alert } from "./Alert";
declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./Alert").AlertProps, import("@chakra-ui/system/dist/system.types").As>>;
export default meta;
export declare const TemplateWithCustomContent: StoryFn<typeof Alert>;
export declare const Error: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./Alert").AlertProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const Info: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./Alert").AlertProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const Success: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./Alert").AlertProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const Warning: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./Alert").AlertProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const Loading: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./Alert").AlertProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const ErrorLeftAccent: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./Alert").AlertProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const WarningLeftAccent: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./Alert").AlertProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const InfoLeftAccent: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./Alert").AlertProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const SuccessLeftAccent: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./Alert").AlertProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const LoadingLeftAccent: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./Alert").AlertProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const CustomChildren: () => JSX.Element;
