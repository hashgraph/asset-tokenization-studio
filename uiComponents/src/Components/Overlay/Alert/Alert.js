var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { Alert as ChakraAlert, AlertDescription as ChakraAlertDescription, } from "@chakra-ui/alert";
import { Box as ChakraBox, Flex as ChakraFlex, Center as ChakraCenter, } from "@chakra-ui/layout";
import { forwardRef, useMultiStyleConfig as useChakraMultiStyleConfig, } from "@chakra-ui/system";
import React from "react";
import { Text } from "@/Components/Foundations/Text";
import { CloseButton } from "@Components/Interaction/CloseButton";
import { PhosphorIcon } from "@/Components/Foundations/PhosphorIcon";
export var alertPartsList = [
    "container",
    "contentContainer",
    "title",
    "description",
    "icon",
    "spinner",
    "closeBtn",
];
export var Alert = forwardRef(function (_a, ref) {
    var children = _a.children, _b = _a.status, status = _b === void 0 ? "info" : _b, _c = _a.showIcon, showIcon = _c === void 0 ? true : _c, _d = _a.isInline, isInline = _d === void 0 ? true : _d, onClose = _a.onClose, variant = _a.variant, props = __rest(_a, ["children", "status", "showIcon", "isInline", "onClose", "variant"]);
    var styles = useChakraMultiStyleConfig("Alert", {
        variant: variant,
        status: status
    });
    var _e = styles.icon, weight = _e.weight, IconComponent = _e.as, restIconStyles = __rest(_e, ["weight", "as"]);
    var Icon = React.useMemo(function () {
        var _a;
        return status === "loading" ? (React.createElement(ChakraCenter, { sx: restIconStyles }, React.cloneElement(IconComponent, {
            size: "xxs",
            color: (_a = restIconStyles["__css"]) === null || _a === void 0 ? void 0 : _a.color
        }))) : (React.createElement(PhosphorIcon, __assign({}, styles.icon, { weight: weight })));
    }, [status, restIconStyles, IconComponent, weight, styles.icon]);
    return (React.createElement(ChakraAlert, __assign({ ref: ref, status: status }, props, { sx: styles.container }),
        React.createElement(ChakraFlex, { w: "full", h: "full", sx: styles.contentContainer },
            showIcon && Icon,
            children ? (React.createElement(ChakraBox, { flex: 1 }, children)) : (React.createElement(ChakraFlex, { "data-testid": "alert-content", direction: isInline ? "row" : "column", flex: 1 },
                React.createElement(ChakraAlertDescription, { sx: styles.description },
                    React.createElement(Text, { as: "span" },
                        props.title && (React.createElement(Text, { as: "span", sx: styles.title }, props.title)),
                        props.description))))),
        onClose && (React.createElement(CloseButton, { sx: styles.closeBtn, onClick: onClose, size: "xs" }))));
});
