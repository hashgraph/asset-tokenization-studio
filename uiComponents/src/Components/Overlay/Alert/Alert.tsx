import type { AlertProps as ChakraAlertProps } from "@chakra-ui/alert";
import {
  Alert as ChakraAlert,
  AlertDescription as ChakraAlertDescription,
} from "@chakra-ui/alert";
import type { alertAnatomy as ChakraAlertParts } from "@chakra-ui/anatomy";
import {
  Box as ChakraBox,
  Flex as ChakraFlex,
  Center as ChakraCenter,
} from "@chakra-ui/layout";
import type { ComponentWithAs, SystemStyleObject } from "@chakra-ui/system";
import {
  forwardRef,
  useMultiStyleConfig as useChakraMultiStyleConfig,
} from "@chakra-ui/system";
import React from "react";
import type {
  BaseMultiStyleConfiguration,
  PartsStyleInterpolation,
} from "@Theme/Components/BaseMultiStyleConfiguration";
import { Text } from "@/Components/Foundations/Text";
import { CloseButton } from "@Components/Interaction/CloseButton";
import { PhosphorIcon } from "@/Components/Foundations/PhosphorIcon";
import type { PhosphorIconProps } from "@/Components/Foundations/PhosphorIcon";

type UnionKeys<T> = T extends T ? keyof T : never;
type StrictUnionHelper<T, TAll> = T extends any
  ? T & Partial<Record<Exclude<UnionKeys<TAll>, keyof T>, never>>
  : never;

type StrictUnion<T> = StrictUnionHelper<T, T>;
export interface BaseAlertProps
  extends Omit<ChakraAlertProps, "children" | "title" | "status"> {
  isInline?: boolean;
  onClose?(): void;
  showIcon?: boolean;
  status?: AlertStatus;
}
export interface CustomChildrenAlertProps {
  children: React.ReactNode;
}

export interface TitleDescriptionAlertProps {
  title?: string;
  description?: string;
}

export type AlertProps = BaseAlertProps &
  StrictUnion<TitleDescriptionAlertProps | CustomChildrenAlertProps>;

export const alertPartsList: Array<
  (typeof ChakraAlertParts.keys)[number] | "closeBtn" | "contentContainer"
> = [
  "container",
  "contentContainer",
  "title",
  "description",
  "icon",
  "spinner",
  "closeBtn",
];

type Parts = typeof alertPartsList;

export type AlertStatus = "success" | "warning" | "error" | "info" | "loading";

export interface AlertThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {
  baseStyle:
    | (({ status }: { status: AlertStatus }) => Partial<AlertThemeStyle>)
    | PartsStyleInterpolation<Parts>;
}

type AlertThemeStyle = Record<Parts[number], SystemStyleObject> & {
  icon: PhosphorIconProps;
};

export const Alert: ComponentWithAs<"div", AlertProps> = forwardRef<
  AlertProps,
  "div"
>(
  (
    {
      children,
      status = "info",
      showIcon = true,
      isInline = true,
      onClose,
      variant,
      ...props
    }: AlertProps,
    ref
  ) => {
    const styles = useChakraMultiStyleConfig("Alert", {
      variant,
      status,
    }) as AlertThemeStyle;

    const { weight, as: IconComponent, ...restIconStyles } = styles.icon;
    const Icon = React.useMemo(
      () =>
        status === "loading" ? (
          <ChakraCenter sx={restIconStyles}>
            {React.cloneElement(IconComponent, {
              size: "xxs",
              color: restIconStyles["__css"]?.color,
            })}
          </ChakraCenter>
        ) : (
          <PhosphorIcon {...styles.icon} weight={weight} />
        ),
      [status, restIconStyles, IconComponent, weight, styles.icon]
    );

    return (
      <ChakraAlert ref={ref} status={status} {...props} sx={styles.container}>
        <ChakraFlex w="full" h="full" sx={styles.contentContainer}>
          {showIcon && Icon}
          {children ? (
            <ChakraBox flex={1}>{children}</ChakraBox>
          ) : (
            <ChakraFlex
              data-testid="alert-content"
              direction={isInline ? "row" : "column"}
              flex={1}
            >
              <ChakraAlertDescription sx={styles.description}>
                <Text as="span">
                  {props.title && (
                    <Text as="span" sx={styles.title}>
                      {props.title}
                    </Text>
                  )}
                  {props.description}
                </Text>
              </ChakraAlertDescription>
            </ChakraFlex>
          )}
        </ChakraFlex>
        {onClose && (
          <CloseButton sx={styles.closeBtn} onClick={onClose} size={"xs"} />
        )}
      </ChakraAlert>
    );
  }
);
