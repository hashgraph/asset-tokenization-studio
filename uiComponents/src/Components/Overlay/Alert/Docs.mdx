import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as AlertStories from "./Alert.stories";

<Meta of={AlertStories} />

<Title />

<a
  href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/ioBricks-Design-System?type=design&node-id=1880%3A17436&mode=design&t=wq5I3vvoIx0bOptY-1"
  target="\_blank"
>
  View figma design
</a>

<Primary />

## How to set up

### Config theme

Add the following configurations to the `theme.tsx` and `Alert.ts` files, respectively.

```tsx dark
const theme = {
  components: {
    Alert: ConfigAlert,
  },
};
```

```tsx dark
const ConfigAlert: AlertThemeConfiguration = {
  parts: alertPartsList,
  baseStyle: {},
};
```

### Component parts

You have these parts available: `"container" | "contentContainer" | "title" | "description" | "icon" | "spinner" | "closeBtn` (`alertPartsList`)

## Props

<ArgTypes of={AlertStories} />

## Stories

<Stories title="" />
