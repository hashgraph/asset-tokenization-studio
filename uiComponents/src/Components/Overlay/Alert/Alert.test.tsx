import { render } from "@/test-utils";
import { Text } from "@chakra-ui/react";
import { waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";
import type { AlertProps, AlertStatus } from "./Alert";
import { Alert } from "./Alert";

describe(`<Alert />`, () => {
  const defaultProps: AlertProps = {
    title: "Title",
    description: "Description",
  };
  const statusList: AlertStatus[] = ["success", "info", "error", "warning"];
  const baseVariantList = ["subtle", "solid", "leftAccent"];
  const factoryComponent = (props: AlertProps = defaultProps) =>
    render(<Alert {...props} />);

  baseVariantList.forEach((variant) => {
    statusList.forEach((status) => {
      test(`renders correctly ${status} alert for variant ${variant}`, () => {
        const component = factoryComponent({
          ...defaultProps,
          status,
          variant,
        });

        expect(component.asFragment()).toMatchSnapshot(
          `Alert-${variant}-${status}`
        );
      });
    });
  });

  test("shows correctly custom children", () => {
    const text = "Example children";
    const component = factoryComponent({ children: <Text>{text}</Text> });

    expect(component.getByText(text)).toBeVisible();
  });

  test("should hide icon when showIcon is false", () => {
    const component = factoryComponent({
      ...defaultProps,
      showIcon: false,
    });

    expect(component.queryByTestId("alert-icon")).not.toBeInTheDocument();
  });

  test("should call onClose when closeBtn is clicked", async () => {
    const onClose = jest.fn();
    const component = factoryComponent({
      ...defaultProps,
      onClose,
    });

    userEvent.click(component.getByLabelText("close-button"));

    await waitFor(() => {
      expect(onClose).toHaveBeenCalled();
    });
  });

  test("should show content inline when isInline is true", async () => {
    const component = factoryComponent({
      ...defaultProps,
      isInline: true,
    });

    const alertContentStyle = getComputedStyle(
      component.getByTestId("alert-content")
    );
    expect(alertContentStyle.flexDirection).toBe("row");
  });
});
