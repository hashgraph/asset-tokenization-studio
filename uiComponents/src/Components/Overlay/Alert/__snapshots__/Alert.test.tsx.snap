// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> renders correctly error alert for variant leftAccent: Alert-leftAccent-error 1`] = `
<DocumentFragment>
  <div
    class="chakra-alert css-lj16js"
    data-status="error"
    description="Description"
    role="alert"
    title="Title"
  >
    <div
      class="css-v4b4ni"
    >
      <svg
        class="chakra-icon css-1wjr3o"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm-8,56a8,8,0,0,1,16,0v56a8,8,0,0,1-16,0Zm8,104a12,12,0,1,1,12-12A12,12,0,0,1,128,184Z"
        />
      </svg>
      <div
        class="css-17fo2je"
        data-testid="alert-content"
      >
        <div
          class="chakra-alert__desc css-1vw6a5w"
          data-status="error"
        >
          <span
            class="css-0"
          >
            <span
              class="css-140t2uq"
            >
              Title
            </span>
            Description
          </span>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Alert /> renders correctly error alert for variant solid: Alert-solid-error 1`] = `
<DocumentFragment>
  <div
    class="chakra-alert css-xuzb86"
    data-status="error"
    description="Description"
    role="alert"
    title="Title"
  >
    <div
      class="css-v4b4ni"
    >
      <svg
        class="chakra-icon css-1wjr3o"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm-8,56a8,8,0,0,1,16,0v56a8,8,0,0,1-16,0Zm8,104a12,12,0,1,1,12-12A12,12,0,0,1,128,184Z"
        />
      </svg>
      <div
        class="css-17fo2je"
        data-testid="alert-content"
      >
        <div
          class="chakra-alert__desc css-hxe6q3"
          data-status="error"
        >
          <span
            class="css-0"
          >
            <span
              class="css-1zbvwl"
            >
              Title
            </span>
            Description
          </span>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Alert /> renders correctly error alert for variant subtle: Alert-subtle-error 1`] = `
<DocumentFragment>
  <div
    class="chakra-alert css-1bfsa0s"
    data-status="error"
    description="Description"
    role="alert"
    title="Title"
  >
    <div
      class="css-v4b4ni"
    >
      <svg
        class="chakra-icon css-1wjr3o"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm-8,56a8,8,0,0,1,16,0v56a8,8,0,0,1-16,0Zm8,104a12,12,0,1,1,12-12A12,12,0,0,1,128,184Z"
        />
      </svg>
      <div
        class="css-17fo2je"
        data-testid="alert-content"
      >
        <div
          class="chakra-alert__desc css-hxe6q3"
          data-status="error"
        >
          <span
            class="css-0"
          >
            <span
              class="css-1zbvwl"
            >
              Title
            </span>
            Description
          </span>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Alert /> renders correctly info alert for variant leftAccent: Alert-leftAccent-info 1`] = `
<DocumentFragment>
  <div
    class="chakra-alert css-d1atnv"
    data-status="info"
    description="Description"
    role="alert"
    title="Title"
  >
    <div
      class="css-v4b4ni"
    >
      <svg
        class="chakra-icon css-1wjr3o"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm-4,48a12,12,0,1,1-12,12A12,12,0,0,1,124,72Zm12,112a16,16,0,0,1-16-16V128a8,8,0,0,1,0-16,16,16,0,0,1,16,16v40a8,8,0,0,1,0,16Z"
        />
      </svg>
      <div
        class="css-17fo2je"
        data-testid="alert-content"
      >
        <div
          class="chakra-alert__desc css-1vw6a5w"
          data-status="info"
        >
          <span
            class="css-0"
          >
            <span
              class="css-140t2uq"
            >
              Title
            </span>
            Description
          </span>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Alert /> renders correctly info alert for variant solid: Alert-solid-info 1`] = `
<DocumentFragment>
  <div
    class="chakra-alert css-1e0we6r"
    data-status="info"
    description="Description"
    role="alert"
    title="Title"
  >
    <div
      class="css-v4b4ni"
    >
      <svg
        class="chakra-icon css-1wjr3o"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm-4,48a12,12,0,1,1-12,12A12,12,0,0,1,124,72Zm12,112a16,16,0,0,1-16-16V128a8,8,0,0,1,0-16,16,16,0,0,1,16,16v40a8,8,0,0,1,0,16Z"
        />
      </svg>
      <div
        class="css-17fo2je"
        data-testid="alert-content"
      >
        <div
          class="chakra-alert__desc css-hxe6q3"
          data-status="info"
        >
          <span
            class="css-0"
          >
            <span
              class="css-1zbvwl"
            >
              Title
            </span>
            Description
          </span>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Alert /> renders correctly info alert for variant subtle: Alert-subtle-info 1`] = `
<DocumentFragment>
  <div
    class="chakra-alert css-mmdfrd"
    data-status="info"
    description="Description"
    role="alert"
    title="Title"
  >
    <div
      class="css-v4b4ni"
    >
      <svg
        class="chakra-icon css-1wjr3o"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm-4,48a12,12,0,1,1-12,12A12,12,0,0,1,124,72Zm12,112a16,16,0,0,1-16-16V128a8,8,0,0,1,0-16,16,16,0,0,1,16,16v40a8,8,0,0,1,0,16Z"
        />
      </svg>
      <div
        class="css-17fo2je"
        data-testid="alert-content"
      >
        <div
          class="chakra-alert__desc css-hxe6q3"
          data-status="info"
        >
          <span
            class="css-0"
          >
            <span
              class="css-1zbvwl"
            >
              Title
            </span>
            Description
          </span>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Alert /> renders correctly success alert for variant leftAccent: Alert-leftAccent-success 1`] = `
<DocumentFragment>
  <div
    class="chakra-alert css-b1zacp"
    data-status="success"
    description="Description"
    role="alert"
    title="Title"
  >
    <div
      class="css-v4b4ni"
    >
      <svg
        class="chakra-icon css-1wjr3o"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm45.66,85.66-56,56a8,8,0,0,1-11.32,0l-24-24a8,8,0,0,1,11.32-11.32L112,148.69l50.34-50.35a8,8,0,0,1,11.32,11.32Z"
        />
      </svg>
      <div
        class="css-17fo2je"
        data-testid="alert-content"
      >
        <div
          class="chakra-alert__desc css-1vw6a5w"
          data-status="success"
        >
          <span
            class="css-0"
          >
            <span
              class="css-140t2uq"
            >
              Title
            </span>
            Description
          </span>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Alert /> renders correctly success alert for variant solid: Alert-solid-success 1`] = `
<DocumentFragment>
  <div
    class="chakra-alert css-6rxtqw"
    data-status="success"
    description="Description"
    role="alert"
    title="Title"
  >
    <div
      class="css-v4b4ni"
    >
      <svg
        class="chakra-icon css-1wjr3o"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm45.66,85.66-56,56a8,8,0,0,1-11.32,0l-24-24a8,8,0,0,1,11.32-11.32L112,148.69l50.34-50.35a8,8,0,0,1,11.32,11.32Z"
        />
      </svg>
      <div
        class="css-17fo2je"
        data-testid="alert-content"
      >
        <div
          class="chakra-alert__desc css-hxe6q3"
          data-status="success"
        >
          <span
            class="css-0"
          >
            <span
              class="css-1zbvwl"
            >
              Title
            </span>
            Description
          </span>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Alert /> renders correctly success alert for variant subtle: Alert-subtle-success 1`] = `
<DocumentFragment>
  <div
    class="chakra-alert css-91ivpr"
    data-status="success"
    description="Description"
    role="alert"
    title="Title"
  >
    <div
      class="css-v4b4ni"
    >
      <svg
        class="chakra-icon css-1wjr3o"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm45.66,85.66-56,56a8,8,0,0,1-11.32,0l-24-24a8,8,0,0,1,11.32-11.32L112,148.69l50.34-50.35a8,8,0,0,1,11.32,11.32Z"
        />
      </svg>
      <div
        class="css-17fo2je"
        data-testid="alert-content"
      >
        <div
          class="chakra-alert__desc css-hxe6q3"
          data-status="success"
        >
          <span
            class="css-0"
          >
            <span
              class="css-1zbvwl"
            >
              Title
            </span>
            Description
          </span>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Alert /> renders correctly warning alert for variant leftAccent: Alert-leftAccent-warning 1`] = `
<DocumentFragment>
  <div
    class="chakra-alert css-rnpyu8"
    data-status="warning"
    description="Description"
    role="alert"
    title="Title"
  >
    <div
      class="css-v4b4ni"
    >
      <svg
        class="chakra-icon css-1wjr3o"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M236.8,188.09,149.35,36.22h0a24.76,24.76,0,0,0-42.7,0L19.2,188.09a23.51,23.51,0,0,0,0,23.72A24.35,24.35,0,0,0,40.55,224h174.9a24.35,24.35,0,0,0,21.33-12.19A23.51,23.51,0,0,0,236.8,188.09ZM120,104a8,8,0,0,1,16,0v40a8,8,0,0,1-16,0Zm8,88a12,12,0,1,1,12-12A12,12,0,0,1,128,192Z"
        />
      </svg>
      <div
        class="css-17fo2je"
        data-testid="alert-content"
      >
        <div
          class="chakra-alert__desc css-1vw6a5w"
          data-status="warning"
        >
          <span
            class="css-0"
          >
            <span
              class="css-140t2uq"
            >
              Title
            </span>
            Description
          </span>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Alert /> renders correctly warning alert for variant solid: Alert-solid-warning 1`] = `
<DocumentFragment>
  <div
    class="chakra-alert css-1midt2v"
    data-status="warning"
    description="Description"
    role="alert"
    title="Title"
  >
    <div
      class="css-v4b4ni"
    >
      <svg
        class="chakra-icon css-1wjr3o"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M236.8,188.09,149.35,36.22h0a24.76,24.76,0,0,0-42.7,0L19.2,188.09a23.51,23.51,0,0,0,0,23.72A24.35,24.35,0,0,0,40.55,224h174.9a24.35,24.35,0,0,0,21.33-12.19A23.51,23.51,0,0,0,236.8,188.09ZM120,104a8,8,0,0,1,16,0v40a8,8,0,0,1-16,0Zm8,88a12,12,0,1,1,12-12A12,12,0,0,1,128,192Z"
        />
      </svg>
      <div
        class="css-17fo2je"
        data-testid="alert-content"
      >
        <div
          class="chakra-alert__desc css-hxe6q3"
          data-status="warning"
        >
          <span
            class="css-0"
          >
            <span
              class="css-1zbvwl"
            >
              Title
            </span>
            Description
          </span>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Alert /> renders correctly warning alert for variant subtle: Alert-subtle-warning 1`] = `
<DocumentFragment>
  <div
    class="chakra-alert css-t03byx"
    data-status="warning"
    description="Description"
    role="alert"
    title="Title"
  >
    <div
      class="css-v4b4ni"
    >
      <svg
        class="chakra-icon css-1wjr3o"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M236.8,188.09,149.35,36.22h0a24.76,24.76,0,0,0-42.7,0L19.2,188.09a23.51,23.51,0,0,0,0,23.72A24.35,24.35,0,0,0,40.55,224h174.9a24.35,24.35,0,0,0,21.33-12.19A23.51,23.51,0,0,0,236.8,188.09ZM120,104a8,8,0,0,1,16,0v40a8,8,0,0,1-16,0Zm8,88a12,12,0,1,1,12-12A12,12,0,0,1,128,192Z"
        />
      </svg>
      <div
        class="css-17fo2je"
        data-testid="alert-content"
      >
        <div
          class="chakra-alert__desc css-hxe6q3"
          data-status="warning"
        >
          <span
            class="css-0"
          >
            <span
              class="css-1zbvwl"
            >
              Title
            </span>
            Description
          </span>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
